geo/src/algorithm/line_interpolate_point.rs

```diff
@@ -24,35 +29,51 @@ use crate::{
 ///     [0.0, 1.0]
 /// ].into();
 ///
-/// assert_eq!(linestring.line_interpolate_point(&-1.0), point!(x: -1.0, y: 0.0));
-/// assert_eq!(linestring.line_interpolate_point(&0.25), point!(x: -0.5, y: 0.0));
-/// assert_eq!(linestring.line_interpolate_point(&0.5), point!(x: 0.0, y: 0.0));
-/// assert_eq!(linestring.line_interpolate_point(&0.75), point!(x: 0.0, y: 0.5));
-/// assert_eq!(linestring.line_interpolate_point(&2.0), point!(x: 0.0, y: 1.0));
+/// assert_eq!(linestring.line_interpolate_point(&-1.0), Some(point!(x: -1.0, y: 0.0)));
+/// assert_eq!(linestring.line_interpolate_point(&0.25), Some(point!(x: -0.5, y: 0.0)));
+/// assert_eq!(linestring.line_interpolate_point(&0.5), Some(point!(x: 0.0, y: 0.0)));
+/// assert_eq!(linestring.line_interpolate_point(&0.75), Some(point!(x: 0.0, y: 0.5)));
+/// assert_eq!(linestring.line_interpolate_point(&2.0), Some(point!(x: 0.0, y: 1.0)));
 /// ```
 pub trait LineInterpolatePoint<F: Float> {
     type Output;
 
     fn line_interpolate_point(&self, fraction: &F) -> Self::Output;
 }
 
+
 impl<T> LineInterpolatePoint<T> for Line<T>
 where
     T: CoordinateType + Float + Zero + One,
 {
-    type Output = Point<T>;
+    type Output = Option<Point<T>>;
 
     fn line_interpolate_point(&self, fraction: &T) -> Self::Output {
-        if fraction < &T::zero() {
-            return self.start.into();
-        };
-        if fraction > &T::one() {
-            return self.end.into();
+        match fraction.partial_cmp(&T::zero()) {
+            Some(o) => match o {
+                Ordering::Less => return Some(self.start.into()),
+                Ordering::Equal => return Some(self.start.into()),
+                Ordering::Greater => {
+                    match fraction.partial_cmp(&T::one()) {
+                        Some(p) => match p {
+                            Ordering::Greater => return Some(self.end.into()),
+                            Ordering::Equal => return Some(self.end.into()),
+                            Ordering::Less => {}
+                        },
+                        None => return None
+                    }
+                }
+            },
+            None => return None
         };
```

Comment:
you can use `?` here to flatten this a bit

```suggestion
        match fraction.partial_cmp(&T::zero())? {
            Ordering::Less => return Some(self.start.into()),
            Ordering::Equal => return Some(self.start.into()),
            Ordering::Greater => {
                match fraction.partial_cmp(&T::one())? {
                    Ordering::Greater => return Some(self.end.into()),
                    Ordering::Equal => return Some(self.end.into()),
                    Ordering::Less => {}
                }
            }
        };
```